singleton-service/src/service/proxy.rs

```diff
@@ -0,0 +1,155 @@
+pub use crate::configuration::service::ServiceConfig;
+pub use crate::service::deltas::DeltaStore;
+use log::{debug, info, warn};
+use proxy_wasm::{
+    hostcalls::{dequeue_shared_queue, register_shared_queue},
+    traits::{Context, RootContext},
+    types::LogLevel,
+};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use threescale::{
+    proxy::cache::{get_application_from_cache, set_application_to_cache},
+    structs::ThreescaleData,
+    utils::update_metrics,
+};
+
+// QUEUE_NAME should be the same as the one in cache filter.
+const QUEUE_NAME: &str = "message_queue";
+
+#[no_mangle]
+pub fn _start() {
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_root_context(|context_id| -> Box<dyn RootContext> {
+        Box::new(SingletonService {
+            context_id,
+            config: ServiceConfig::default(),
+            queue_id: None,
+            delta_store: DeltaStore {
+                last_update: None,
+                request_count: 0,
+                deltas: RefCell::new(HashMap::new()),
+            },
+        })
+    });
+}
+
+struct SingletonService {
+    context_id: u32,
+    config: ServiceConfig,
+    queue_id: Option<u32>,
+    delta_store: DeltaStore,
+}
+
+impl RootContext for SingletonService {
+    /// Message queue will get registered when on_vm_start callback gets called.
+    fn on_vm_start(&mut self, _vm_configuration_size: usize) -> bool {
+        if let Ok(q_id) = register_shared_queue(QUEUE_NAME) {
+            self.queue_id = Some(q_id);
+        }
+        // TODO : handle MQ failure, change info to trace after dev
+        info!(
+            "Registered new message queue with id: {}",
+            self.queue_id.unwrap()
+        );
+        true
+    }
+
+    /// Configuration passed by envoy.yaml will get deserialized to ServiceConfig. If there's an issue with the
+    /// passed configuration, default configuration will be used.
+    fn on_configure(&mut self, _config_size: usize) -> bool {
+        // Check for the configuration passed by envoy.yaml
+        let configuration: Vec<u8> = match self.get_configuration() {
+            Some(c) => c,
+            None => {
+                warn!("Configuration missing. Please check the envoy.yaml file for filter configuration");
+                return false;
```

Comment:
We already have a default config, so we don't have to cause a panic if the configuration is missing. Please set this to 'true`.

Reply:
I was using the default config for the scenario of config provided and parsing fails. But yeah even for the scenario where the user hasn't provided a config, we can go with the default.